io/src/lib.rs

```diff
@@ -35,16 +31,40 @@ pub struct InvariantConfig {
     pub protocol_fee: u128,
 }
 
-#[derive(Clone, Decode, Encode, TypeInfo)]
+#[derive(Clone, Decode, Encode, Debug, TypeInfo)]
 #[codec(crate = gstd::codec)]
 #[scale_info(crate = gstd::scale_info)]
 pub enum InvariantAction {
     ChangeProtocolFee(u128),
+    AddFeeTier(FeeTier),
+    RemoveFeeTier(FeeTier)
 }
 
-#[derive(Clone, Decode, Encode, TypeInfo)]
+#[derive(Clone, Decode, Encode, Debug, PartialEq, Eq, TypeInfo)]
 #[codec(crate = gstd::codec)]
 #[scale_info(crate = gstd::scale_info)]
 pub enum InvariantEvent {
     ProtocolFeeChanged(u128),
+    FeeTierAdded(FeeTier),
```

Comment:
We don't need events for everything, just these ones that will be using for statistics. The events related to fee tiers will not be utilized, so it's worth saving resources